### PR TITLE
Rebased geojson

### PR DESCRIPTION
**Pull request by robertbartel**
Added parsing capabilities for Geojson documents (this is a rebased version of the changes proposed in PR #36 along with a few additional directly related improvements).

## Additions

- Handling for Geojson documents

## Testing

1. Run the `test_geojson`, `test_unit`, or `test_all` target to test the functionality.  Keep in mind however that, except for `test_geojson`, other unrelated tests will also run.

## Notes

- Handling for GeometryCollections is not complete
- In PR #36, build issues were keeping everything from working if the implementation code wasn't in the header. That should be addressed for the most part in this PR.

## Todos

- Complete the documentation

### CODE REVIEW

File: src/geojson/Makefile [hunk 1]
```diff
@@ -0,0 +1,332 @@
+# CMAKE generated file: DO NOT EDIT!
```
Comment (hellkite500) [inline src/geojson/Makefile]: Generated files should not be indexed and versioned.

Comment (hellkite500) [reply src/geojson/Makefile]: Similarly,  src/geojson/libgeojson.a should get removed.

**Comment by robertbartel** [comment]
Indeed, I didn't notice those had snuck in.  I'll remove them.  I think I also want to fix a couple other things also ...

**Comment by robertbartel** [comment]
Ok @hellkite500 and @christophertubbs:  I have those cleaned up, and have actually moved code definitions to code files for the rest of the classes.  I may not have been entirely consistent about how I did this with constructors, so those may need to be modified (as part of this PR, or later).

File: include/geojson/Feature.hpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            );
+
+            Feature(std::vector<JSONGeometry> new_geometry_collection,
+                std::vector<double> new_bounding_box = std::vector<double>(),
+                property_map new_properties = property_map()
+            );
+
+            Feature(boost::property_tree::ptree &tree);
+
+            Feature(const Feature &feature);
```
Comment (hellkite500) [inline include/geojson/Feature.hpp:43]: If we are going to have a copy constructor, I would advocate for a swap function as well.  Copy Swap is a nice pattern for implementing various semantics, and might be a good idea to put in as many places as possible.  If performance issues arise we can profile and tackle them on an individual basis.

File: include/geojson/JSONGeometry.hpp [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+            static JSONGeometry of_point(pt::ptree &tree);
+
+            static JSONGeometry of_linestring(two_dimensional_coordinates &coordinates);
+
+            static JSONGeometry of_linestring(pt::ptree &tree);
+
+            static JSONGeometry of_polygon(three_dimensional_coordinates &coordinates);
+
+            /**
+             * Generate a polyginal geometry from the details of coordinates passed in shaped like:
```
Comment (hellkite500) [inline include/geojson/JSONGeometry.hpp:138]: polygonal?

File: src/geojson/Feature.cpp [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+            if (property.second.data() == "true") {
+                properties.emplace(key, JSONProperty(key, true));
+                continue;
+            }
+            else if (property.second.data() == "false") {
+                properties.emplace(key, JSONProperty(key, false));
+                continue;
+            }
+
+            for(int character_index = 0; character_index < value.length(); character_index++) {
```
Comment (hellkite500) [inline src/geojson/Feature.cpp:110]: I would be inclined to move this character parsing to a private function by itself.

File: include/geojson/JSONGeometry.hpp [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+            bg::model::multi_linestring<linestring_t> as_multilinestring() const;
+
+            bg::model::multi_polygon<polygon_t> as_multipolygon() const;
+        private:
+            JSONGeometryType type;
+            linestring_t linestring;
+            coordinate_t point;
+            polygon_t polygon;
+            bg::model::multi_point<coordinate_t> multi_point;
```
Comment (hellkite500) [inline include/geojson/JSONGeometry.hpp:197]: Need to think about a better way to dynamically type the boost geom model.

**Comment by hellkite500** [comment]
@christophertubbs @robertbartel I wanted to give you a chance to see the comments in the review I posted.  I want to merge this as a starting point to iterate on unless you think there is anything we should address out of those comments

**Pull request merged**